mypyc/irbuild/ll_builder.py

```diff
@@ -779,8 +779,12 @@ def call_c(self,
         # add extra integer constant if any
         if desc.extra_int_constant is not None:
             val, typ = desc.extra_int_constant
-            extra_int_constant = self.add(LoadInt(val, line, rtype=typ))
-            coerced.append(extra_int_constant)
+            if is_pointer_rprimitive(typ) and val == 0:
+                casted_constant = self.add(LoadGlobal(object_rprimitive, "NULL"))
+                coerced.append(casted_constant)
+            else:
+                extra_int_constant = self.add(LoadInt(val, line, rtype=typ))
+                coerced.append(extra_int_constant)
```

Comment:
Here's the special case. @JukkaL what do you think? If we decide this is a workable approach, I'll first fix the `to_str` of `LoadGlobal` since there's a hack there to ignore the prefix. I'll remove that hack and have all the related IR changes in a separate PR and merge this PR with master.

Reply:
I'm requesting a review for this even the build will fail.

Reply:
This looks good.

Reply:
> This looks good.

Glad to here that. I've submitted #9331 before updating the IR tests in this PR.

Reply:
Hmm I would actually prefer to do the translation to NULL in `mypy.codegen`, since it's only relevant for the C back end. We can represent null pointers as integer 0 values in the IR, but we should use NULL in the generated C.